app/ante/check_validator_decorator.go

```diff
@@ -0,0 +1,68 @@
+package ante
+
+import (
+	errorsmod "cosmossdk.io/errors"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/planetmint/planetmint-go/x/dao/types"
+)
+
+type CheckValidatorDecorator struct {
+	sk StakingKeeper
+}
+
+func NewCheckValidatorDecorator(sk StakingKeeper) CheckValidatorDecorator {
+	return CheckValidatorDecorator{sk: sk}
+}
+
+func (cv CheckValidatorDecorator) AnteHandle(ctx sdk.Context, tx sdk.Tx, simulate bool, next sdk.AnteHandler) (_ sdk.Context, err error) {
+	if simulate || ctx.BlockHeight() == 0 {
+		return next(ctx, tx, simulate)
+	}
+
+	for _, msg := range tx.GetMsgs() {
```

Comment:
could this be re-written to?
```
        for _, msg := range tx.GetMsgs() {
                switch sdk.MsgTypeURL(msg) {
                case "/planetmintgo.dao.MsgInitPop":
                        fallthrough
                case "/planetmintgo.dao.MsgDistributionRequest":
                        fallthrough
                case "/planetmintgo.dao.MsgDistributionResult":
                        fallthrough
                case "/planetmintgo.dao.MsgReissueRDDLProposal":
                        fallthrough
                case "/planetmintgo.dao.MsgReissueRDDLResult":
                        ctx, err = cv.handleMsg(ctx, msg)
                default:
                        continue
                }
        }
```

`cv.handleMsg()` is for all message types the same, so type assertion is unnecessary in my opinion

Reply:
https://golangbyexample.com/fallthrough-keyword-golang/